### DISCUSSION TITLE
[@types/mdx] `Component` should also include `Promise<JSX.Element>`

### DISCUSSION THREAD

**Discussion by inflation** [DefinitelyTyped/DefinitelyTyped#64885] [Issues with a @types package]
Because React Server Component can be async, so `MDXContentProp` should also accept async components.

[a comment by an automated account is not shown]

**Comment by remcohaszing** [comment]
This is caused by a bug in TypeScript. `JSX.Element` represents both the return type of a JSX function component, as well as a the type of a JSX expression. Unfortunately this can’t be fixed by any JSX framework types. This affects the types of React, Preact, Vue, MDX, etc. See https://github.com/microsoft/TypeScript/issues/21699 for details.

**Reply by remcohaszing** [reply]
As a practical example, this is the same reason why TypeScript complains about the following valid React code:

```ts
export function Foo() {
  return 'React allows you to return strings, TypeScript does not'
}

export function Bar() {
  return <Foo />
}
```

**Comment by remcohaszing** [comment]
This was fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65703
